clu/src/lib.rs

```diff
@@ -92,21 +86,65 @@ fn linux_init(config: Arc<RwLock<settings::RitaSettingsStruct>>) -> Result<(), E
     cleanup()?;
     KI.restore_default_route(&mut config.get_network_mut().default_route)?;
 
-    let privkey = config.get_network().wg_private_key.clone();
-    let pubkey = config.get_network().wg_public_key.clone();
-    let mesh_ip = config.get_network().own_ip.clone();
+    let mut network_settings = config.get_network_mut();
```

Comment:
holding a write reference for the duration of this function kinda makes me nervous, but at the same time it's pre-start

Reply:
So normally to me cases like this usually turn out to be a trade-off between atomicity of a bunch of code and the risk of hogging the lock unnecessarily. But here to me it makes sense to lean on the hogging side because the network settings section of the config is potentially useless without `clu` making sure everything's in place.